pycheribuild/projects/disk_image.py

```diff
@@ -676,24 +680,29 @@ def make_disk_image(self):
             else:
                 self.info("qemu-img command was not found. Will not be able to create QCOW2 images")
 
+        if self.use_qcow2:
+            # If we're going to generate a qcow2 image, avoid clobbering the non-qcow2 image and don't generate a .qcow2
+            # file that isn't acqually qcow2.
+            raw_img = self.disk_image_path.with_suffix(".raw")
+        else:
+            raw_img = self.disk_image_path
+
         if self.rootfs_only:
-            self.make_rootfs_image(self.disk_image_path)
+            self.make_rootfs_image(raw_img)
         elif self.is_x86:
             # X86 currently requires special handling
             # TODO: Switch to normal UEFI booting
-            self.make_x86_disk_image()
+            self.make_x86_disk_image(raw_img)
         else:
-            self.make_gpt_disk_image()
+            self.make_gpt_disk_image(raw_img)
 
         # Converting QEMU images: https://en.wikibooks.org/wiki/QEMU/Images
         if not self.config.quiet and qemu_img_command.exists():
-            self.run_cmd(qemu_img_command, "info", self.disk_image_path)
+            self.run_cmd(qemu_img_command, "info", raw_img)
         if self.use_qcow2:
             if not qemu_img_command.exists():
                 self.fatal("Cannot create QCOW2 image without qemu-img command!")
             # create a qcow2 version from the raw image:
-            raw_img = self.disk_image_path.with_suffix(".raw")
-            self.run_cmd("mv", "-f", self.disk_image_path, raw_img)
             self.run_cmd(qemu_img_command, "convert",
                          "-f", "raw",  # input file is in raw format (not required as QEMU can detect it
                          "-O", "qcow2",  # convert to qcow2 format
```

Comment:
Do we want to use zstd compression here?

Reply:
If you'd rather that be here than in the docker goo, I'm happy to do it here.  I was contemplating `--disk-image/use-qcow2=zstd` or something, and so hadn't added it yet, but I'm also OK just always compressing qcow2 images.

Reply:
I guess disk I/O get's a bit slower but I doubt very much that that is limiting our CheriBSD on QEMU performance. Compression also appears to be very quick so I don't think there is a reason not to do it by default? If you don't want it you can always use raw images.

Reply:
OK, I'm sold.